tests/phpunit/tests/option/option.php

```diff
@@ -312,4 +312,254 @@ public function data_option_autoloading() {
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::prime_options_by_group
+	 */
+	public function test_prime_options_by_group() {
+		// Create some options to prime.
+		$new_allowed_options = array(
+			'group1' => array(
+				'option1',
+				'option2',
+			),
+			'group2' => array(
+				'option3',
+			),
+		);
+
+		$options_to_prime = array(
+			'option1',
+			'option2',
+			'option3',
+		);
+
+		/*
+		 * Set values for the options,
+		 * clear the cache for the options,
+		 * check options are not in cache initially.
+		 */
+		foreach ( $options_to_prime as $option ) {
+			update_option( $option, "value_$option", false );
+			wp_cache_delete( $option, 'options' );
+			$this->assertFalse( wp_cache_get( $option, 'options' ), "$option was not deleted from the cache." );
+		}
+
+		// Call the prime_options_by_group function to prime the options.
+		prime_options_by_group( 'group1' );
+
+		// Check that options are now in the cache.
+		$this->assertSame( get_option( 'option1' ), wp_cache_get( 'option1', 'options' ), 'option1 was not primed.' );
+		$this->assertSame( get_option( 'option2' ), wp_cache_get( 'option2', 'options' ), 'option2 was not primed.' );
+
+		// Make sure option3 is still not in cache.
+		$this->assertFalse( wp_cache_get( 'option3', 'options' ), 'option3 was not deleted from the cache.' );
+	}
+
+	/**
+	 * Test get_options function.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::prime_options
+	 * @covers ::get_options
+	 */
+	public function test_get_options() {
+		// Create some options to prime.
+		$options_to_prime = array(
+			'option1',
+			'option2',
+		);
+
+		/*
+		 * Set values for the options,
+		 * clear the cache for the options,
+		 * check options are not in cache initially.
+		 */
+		foreach ( $options_to_prime as $option ) {
+			update_option( $option, "value_$option", false );
+			wp_cache_delete( $option, 'options' );
+			$this->assertFalse( wp_cache_get( $option, 'options' ), "$option was not deleted from the cache." );
+		}
+
+		// Call the prime_options function to prime the options.
+		prime_options( $options_to_prime );
+
+		// Check that options are now in the cache.
+		foreach ( $options_to_prime as $option ) {
+			$this->assertSame( wp_cache_get( $option, 'options' ), get_option( $option ), "$option was not primed." );
+		}
+
+		// Call the get_options function to retrieve the options.
+		$options = get_options( array( 'option1', 'option2' ) );
+
+		// Check that the retrieved options are correct.
+		$this->assertSame( get_option( 'option1' ), $options['option1'] );
+		$this->assertSame( get_option( 'option2' ), $options['option2'] );
+	}
+
+	/**
+	 * Test get_options with an empty input array.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::get_options
+	 */
+	public function test_get_options_with_empty_array() {
+		// Call the get_options function with an empty array.
+		$options = get_options( array() );
+
+		// Make sure the result is an empty array.
+		$this->assertEmpty( $options );
```

Comment:
`assertEmpty()` will pass for _any_ value that passes a PHP `empty()` check, including `(int) 0`, `(float) 0.0`, `(string) "0"`, `(string) ""`, `(bool) false`, `null`, `(array) []`.

To test for an empty array, use:
```php
$this->assertIsArray( $options, 'An array should have been returned.' );
$this->assertEmpty( $options, 'No options should have been returned.' );
```